src/documents/DataSourceAPI.md

```diff
@@ -0,0 +1,135 @@
+---
+title: DataSource API
+category: Infusion
+---
+
+The DataSource component provides a workflow for getting/setting (read/write) data from a data source (an external
+database, cookie, in memory storage and etc). The specifics for accessing the data source must be provided in a concrete
+implementation. Additionally the payload of a request may be configured to be processed through an encoding/decoding
+step and other transformations during the get and set workflows.
+
+## Grades and Linkage
+
+There are two main grades `fluid.dataSource` and `fluid.dataSource.writable`. `fluid.dataSource` contains the base
+configuration and includes configuration for getting (reading) from a data source. `fluid.dataSource.writable` adds the
+configuration for setting (writing) to a data source. Instances of `fluid.dataSource` will need to provide a concrete
+handler for the `"onRead.impl"` and those using `fluid.dataSource.writable` will also need to add one for the
+`"onWrite.impl"` listener.
+
+```javascript
+fluid.defaults("my.component", {
+    components: {
+        dataSource: {
+            type: "fluid.dataSource",
+            options: {
+                gradeNames: ["fluid.dataSource.writable"],
+                listeners: {
+                    // these would point at concrete implementations for performing the read and write operations.
+                    "onRead.impl": "my.component.doRead",
+                    "onWrite.impl": "my.component.doWrite"
+                }
+            }
+        }
+    }
+});
+```
+
+When implementing a new kind of dataSource, where reading and writing functionality are needed,
+[grade linkage](IoCAPI.md#fluidmakegradelinkagelinkagename-inputnames-outputnames) is required to apply the concrete
+writable grade to the dataSource configuration. In this way, one would only need to add the `fluid.dataSource.writable`
```

Comment:
"In this way, one would need" -> "In this way, a user who has already applied the name of a particular concrete read-only dataSource only needs to add `fluid.dataSource.writable` as a further mixin in order to get a read/write implementation, regardless of what the original concrete grade was"

We should think about whether we want to backport the version of this which is in the "Infusion 4" branches which relies instead on a flag `writable: true/false` and uses ContextAwareness rather than grade linkage so that we can avoid API changes in two successive releases

Reply:
I'll update the docs with your suggestions for now. If you it won't be too much work to back port the `writable: true/false` flag, I think it would make sense to do that.

Reply:
@amb26 now https://github.com/fluid-project/infusion/pull/1050 that has been merged, I've updated the docs to reflect the changes for [FLUID-6623](https://issues.fluidproject.org/browse/FLUID-6623).